components/popover_list_members/popover_list_members.tsx

```diff
@@ -0,0 +1,293 @@
+
+import {ModalData} from 'types/actions';
+
+import ChannelMembersModal from 'components/channel_members_modal';
+import ChannelInviteModal from 'components/channel_invite_modal';
+import OverlayTrigger from 'components/overlay_trigger';
+import Tooltip from 'components/tooltip';
+import Popover from 'components/widgets/popover';
+import PopoverListMembersItem from 'components/popover_list_members/popover_list_members_item';
+import {RelationOneToOne} from 'mattermost-redux/types/utilities';
+
+export type Props = {
+    channel: Channel;
+    statuses: RelationOneToOne<UserProfile, string>;
+    users: UserProfile[];
+    memberCount: number;
+    currentUserId: string;
+    manageMembers: boolean;
+    actions: {
+        openModal: <P>(modalData: ModalData<P>) => void;
+        loadProfilesAndStatusesInChannel: (channelId: string, page?: number, perPage?: number, sort?: string, options?: {active: boolean}) => void;
+        openDirectChannelToUserId: (userId: string) => Promise<{data: Channel}>;
+    };
+    sortedUsers: UserProfile[];
+    addMembersABTest?: string;
+    teamUrl?: string;
+}
+
+export default function PopoverListMembers(props: Props) {
+    const membersListRef: React.RefObject<HTMLDivElement> = React.createRef();
+    const targetRef: React.RefObject<HTMLButtonElement> = React.createRef();
+
+    const [showPopover, setShowPopover] = useState(false);
+    const [users, setUsers] = useState<UserProfile[]>(props.users);
+    const [statuses, setStatuses] = useState<RelationOneToOne<UserProfile, string>>(props.statuses);
+
+    useEffect(() => {
+        setUsers(props.users);
+        setStatuses(props.statuses);
+    }, [props.users, props.statuses]);
+
+    const handleShowDirectChannel = (user: UserProfile) => {
+        const {actions} = props;
+        const teammateId = user.id;
+
+        if (teammateId) {
+            actions.openDirectChannelToUserId(teammateId).then(({data}) => {
+                if (data) {
+                    browserHistory.push(props.teamUrl + '/channels/' + data.name);
+                }
+                closePopover();
+            });
+        }
+    };
+
+    const closePopover = () => {
+        setShowPopover(false);
+    };
+
+    const showMembersModal = (e: React.MouseEvent<HTMLButtonElement>) => {
+        e.preventDefault();
+
+        closePopover();
+
+        const modalData = {
+            modalId: ModalIdentifiers.CHANNEL_MEMBERS,
+            dialogProps: props,
+            dialogType: ChannelMembersModal,
+        };
+
+        props.actions.openModal(modalData);
+    };
+
+    const onAddNewMembersButton = (placement: string) => {
+        const {channel, actions} = props;
+        trackEvent('add_members_from_channel_popover', placement);
+
+        actions.openModal({
+            modalId: ModalIdentifiers.CHANNEL_INVITE,
+            dialogType: ChannelInviteModal,
+            dialogProps: {channel},
+        });
+
+        closePopover();
+    };
```

Comment:
these 2 functions are written differently, although they are basically doing the same thing (with differing data). The question is: could we simplify the functions into one with flagging parameters, or at least built them in a way that they look and work the same?

Reply:
yeah, but after thinking bout it for a while I would say it would be better to just have them look the same to make them more readable. Making them work with flags might be too much and make them error-prone as well.

Reply:
I updated the part related to creating `modalData` and calling `actions.openModal(modalData)` to make them look similar. Please let me know if there is more can be done here